cub/cub/util_arch.cuh

```diff
@@ -73,42 +73,14 @@ CUB_NAMESPACE_BEGIN
 #    endif
 #  endif
 
-/// Maximum number of devices supported.
-#  ifndef CUB_MAX_DEVICES
-#    define CUB_MAX_DEVICES (128)
-#  endif
-
-static_assert(CUB_MAX_DEVICES > 0, "CUB_MAX_DEVICES must be greater than 0.");
-
-/// Number of threads per warp
-#  ifndef CUB_LOG_WARP_THREADS
-#    define CUB_LOG_WARP_THREADS(unused) (5)
-#    define CUB_WARP_THREADS(unused)     (1 << CUB_LOG_WARP_THREADS(0))
-
-#    define CUB_PTX_WARP_THREADS     CUB_WARP_THREADS(0)
-#    define CUB_PTX_LOG_WARP_THREADS CUB_LOG_WARP_THREADS(0)
-#  endif
-
-/// Number of smem banks
-#  ifndef CUB_LOG_SMEM_BANKS
-#    define CUB_LOG_SMEM_BANKS(unused) (5)
-#    define CUB_SMEM_BANKS(unused)     (1 << CUB_LOG_SMEM_BANKS(0))
-
-#    define CUB_PTX_LOG_SMEM_BANKS CUB_LOG_SMEM_BANKS(0)
-#    define CUB_PTX_SMEM_BANKS     CUB_SMEM_BANKS
-#  endif
-
-/// Oversubscription factor
-#  ifndef CUB_SUBSCRIPTION_FACTOR
-#    define CUB_SUBSCRIPTION_FACTOR(unused) (5)
-#    define CUB_PTX_SUBSCRIPTION_FACTOR     CUB_SUBSCRIPTION_FACTOR(0)
-#  endif
-
-/// Prefer padding overhead vs X-way conflicts greater than this threshold
-#  ifndef CUB_PREFER_CONFLICT_OVER_PADDING
-#    define CUB_PREFER_CONFLICT_OVER_PADDING(unused) (1)
-#    define CUB_PTX_PREFER_CONFLICT_OVER_PADDING     CUB_PREFER_CONFLICT_OVER_PADDING(0)
-#  endif
+inline constexpr int max_devices       = 128;
+inline constexpr int warp_threads      = 32;
+inline constexpr int log2_warp_threads = 5;
+inline constexpr int smem_banks        = 32;
```

Comment:
I am not sure we want to publicly expose this information again. Is there a specific reason for this? If not, I would prefer those constants to be inside `cub::detail`.